app/views/courses/TeacherCoursesView.coffee

```diff
@@ -99,3 +101,22 @@ module.exports = class TeacherCoursesView extends RootView
     else
       changeLogText.addClass('hidden')
       showChangeLog.text($.i18n.t('courses.show_change_log'))
+
+  onClickVideoThumbnail: (e) ->
+    @$('#video-modal').modal('show')
+    image_src = e.target.src.slice(e.target.src.search('/images'))
+    @$('.video-player')[0].src = (Object.values(@videoLevels || {}).find((l) => l.thumbnail_unlocked == image_src) || {}).url
```

Comment:
`@videoLevels` initialized above, so can it ever be null or undefined?

Reply:
hmm, should not be ideally